pkg/operator/vsphere_check.go

```diff
@@ -305,6 +306,10 @@ func newClient(ctx context.Context, cfg *vsphere.VSphereConfig, username, passwo
 	defer cancel()
 	klog.V(4).Infof("Connecting to %s as %s, insecure %t", serverAddress, username, insecure)
 	client, err := govmomi.NewClient(tctx, serverURL, insecure)
+
+	vpdVersion := version.Get()
+	client.UserAgent = fmt.Sprintf("vsphere-problem-detector/%s", vpdVersion)
+
```

Comment:
@bostrt did you have a chance to test this patch?

Recently I tried to do the same in the vSphere CSI driver operator but `govc sessions.ls` didn't return the new user agent.